otherlibs/runtime_events/runtime_events.mli

```diff
@@ -119,6 +119,49 @@ module Timestamp : sig
     val to_int64 : t -> int64
 end
 
+module Type : sig
+  type 'a t
+  (** The type for an user event content type *)
+
+  val event : unit t
+  (** An event has no data associated with it *)
+
+  val counter : int t
+  (** A counter is a number *)
+
+  val register : ('a -> bytes) -> (bytes -> 'a) -> 'a t
```

Comment:
I would use labels here:

```ocaml
val register:
  encode:('a -> bytes) ->
  decode:(bytes -> 'a) ->
  unit -> 'a t
```

Later, if more fields are useful (e.g. a human readable pretty-printer), they can be added as optional arguments without breaking existing callers.